modules/tensor_mechanics/src/actions/TensorMechanicsRSphericalAction.C

```diff
@@ -0,0 +1,75 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "TensorMechanicsRSphericalAction.h"
+
+#include "Factory.h"
+#include "FEProblem.h"
+#include "Parser.h"
+
+template<>
+InputParameters validParams<TensorMechanicsRSphericalAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("Set up stress divergence kernel for the 1D spherical problem");
+  params.addRequiredParam<std::vector<NonlinearVariableName> >("displacements", "The nonlinear displacement variable for the problem (should only be a single variable)");
+  params.addParam<std::string>("base_name", "Material property base name");
+  params.addParam<bool>("use_displaced_mesh", false, "Whether to use displaced mesh in the kernels");
+
+  params.addParam<std::vector<AuxVariableName> >("save_in_disp_r", "Auxiliary variables to save the r displacement residuals.");
+  return params;
+}
+
+TensorMechanicsRSphericalAction::TensorMechanicsRSphericalAction(const InputParameters & params) :
+    Action(params)
+{
+}
+
+void
+TensorMechanicsRSphericalAction::act()
+{
+  std::vector<NonlinearVariableName> displacements = getParam<std::vector<NonlinearVariableName> > ("displacements");
+  std::vector<VariableName> coupled_displacements;
+  unsigned int dim = displacements.size();
+
+  //Error checking:  Can only take one displacement variables in RSpherical kernel
+  mooseAssert(dim == 1, "Expected a single displacement variable but recieved " << dim);
+
+  for (unsigned int i = 0; i < dim; ++i)
+  {
+    coupled_displacements.push_back(displacements[i]);
+  }
+
+  // Retain this code 'as is' because StressDivergenceTensors inherits from Kernel.C
+  std::vector<std::vector<AuxVariableName> > save_in;
+  save_in.resize(dim);
+
+  if (isParamValid("save_in_disp_r"))
+    save_in[0] = getParam<std::vector<AuxVariableName> >("save_in_disp_r");
+
+  InputParameters params = _factory.getValidParams("StressDivergenceRSphericalTensors");
+  params.set<std::vector<VariableName> >("displacements") = coupled_displacements;
+
+  params.set<bool>("use_displaced_mesh") = getParam<bool>("use_displaced_mesh");
+
+  if (isParamValid("base_name"))
+    params.set<std::string>("base_name") = getParam<std::string>("base_name");
+
+  std::string short_name = "TensorMechanicsRSpherical";
+
+  for (unsigned int i = 0; i < dim; ++i)
+  {
+    std::stringstream name;
+    name << short_name;
+    name << i;
```

Comment:
It's used all over in the framework. Just try grepping through the source
for examples.

$ find . -name "*.[Ch]" | xargs grep stringify

It looks like a normal function call. It turns your argument into a string
that can be concatenated like a string to the rest of your error or warning
message.

Cody

On Mon, Feb 29, 2016 at 9:47 AM Stephanie Pitts <notifications@github.com>
wrote:

> In modules/tensor_mechanics/src/actions/TensorMechanicsRSphericalAction.C
> <https://github.com/idaholab/moose/pull/6380#discussion_r54438409>:
>
> > +
> > +  InputParameters params = _factory.getValidParams("StressDivergenceRSphericalTensors");
> > +  params.set<std::vector<VariableName> >("displacements") = coupled_displacements;
> > +
> > +  params.set<bool>("use_displaced_mesh") = getParam<bool>("use_displaced_mesh");
> > +
> > +  if (isParamValid("base_name"))
> > +    params.set<std::string>("base_name") = getParam<std::string>("base_name");
> > +
> > +  std::string short_name = "TensorMechanicsRSpherical";
> > +
> > +  for (unsigned int i = 0; i < dim; ++i)
> > +  {
> > +    std::stringstream name;
> > +    name << short_name;
> > +    name << i;
>
> Is there a better example for how to use stringify than
> phase_field/src/materials/PolyCrystalKernelAction.C ?
>
> We also use stringstream in the TensorMechanicsAction, PressureAction, and
> PoroMechanicsAction (clearly tensor mechanics missed the memo about
> Moose::stringify() ; would you like these additional actions updated at the
> same time as I make the corrections to the Axisymmetric and RSpherical
> actions?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6380/files#r54438409>.
>